### PR TITLE
Fix regression in `Hash#dig` for HashWithIndifferentAccess.

### PR DESCRIPTION
**Pull request by maclover7**
Fixes issue #23411 

cc @claudiob @sgrif

### CODE REVIEW

File: activesupport/lib/active_support/hash_with_indifferent_access.rb [hunk 1]
```diff
@@ -69,9 +69,13 @@ def initialize(constructor = {})
...
     def default(*args)
-      key = args.first
-      args[0] = key.to_s if key.is_a?(Symbol)
-      super(*args)
+      arg_key = args.first
+
+      if include?(key = convert_key(arg_key))
+        self[key]
+      else
+        super(*args)
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/hash_with_indifferent_access.rb]: Only `super` here.

**Comment by rafaelfranca** [comment]
This is a pref regression compared with David's commit but if you look the [code before David's commit it is also using `include?`](https://github.com/rails/rails/commit/38a990237a0f99d99b885ea6570cae3752138a29#diff-9f4b3d7342097a6f22290f147a09b77bL72) so I think this is fine.

**Comment by maclover7** [comment]
updated.

**Comment by claudiob** [comment]
Thanks y'all are the best! :tada: :tada: :tada: :tada:

**Comment by maclover7** [comment]
:heart:

**Pull request merged**